### ISSUE TITLE
Automatic namespaces ignore manual namespaces

### ISSUE THREAD

**Issue by pawelszramowski** [plantuml/plantuml#632]
This is best described with a class diagram:

```plantuml
@startuml
namespace foo {
    ' c1 will be in foo namespace
    class c1

    ' ACTUAL: c2 will be in bar namespace independent of foo namespace (nesting is ignored)
    ' EXPECTED: c2 will be in bar namespace nested in foo namespace
    class bar.c2
}
@enduml
```

PlantUML 2021.09 and older

**Comment by BenceSzalai** [comment]
Probably it is due to the same root cause as #634.

**Comment by The-Lum** [comment]
Hi @pawelszramowski, @BenceSzalai, and all,

With last changes on namespaces and packages _(see [news](https://plantuml.com/en/news))_: it seems _quasi_ fixed.

```puml
@startuml
namespace foo {
    ' c1 will be in foo namespace
    class c1

    ' EXPECTED: c2 will be in bar namespace nested in foo namespace
    class bar.c2
}
@enduml
```
>[![](https://www.planttext.com/api/plantuml/svg/VSyn2m8n30NWFRyYYwXBzUBcTA3T7HnSqrw4Gg-LfkeW_dVB8UfarlUz1-crSgdNCKBYKVJ2Nl2SCpu0sytHTtWFCQ8J36bAFcuIFh9gK_3szdImkwFTRz7JRzDn-JOnYLOP_YqsRpp14ngeKh0pjAQD8IUL2LSukqdHaDDY2RsaeJtm0W00)](https://www.planttext.com/?text=VSyn2m8n30NWFRyYYwXBzUBcTA3T7HnSqrw4Gg-LfkeW_dVB8UfarlUz1-crSgdNCKBYKVJ2Nl2SCpu0sytHTtWFCQ8J36bAFcuIFh9gK_3szdImkwFTRz7JRzDn-JOnYLOP_YqsRpp14ngeKh0pjAQD8IUL2LSukqdHaDDY2RsaeJtm0W00)


@arnaudroques:
- Perhaps only the name (`foo.bar` instead only `bar` on the header of `bar` package) is now **wrong**...

Regards.

**Comment by BenceSzalai** [comment]
Great to see such a meaningful progress in this aspect!

> Perhaps only the name (foo.bar instead only bar on the header of bar package) is now wrong...

That seems to be a bit odd, hopefully that'll be fixed too. Or maybe a config option should be added for that, as I can imagine that on very large diagrams a fully qualified namespace name is more beneficial, as without it one may need to pan and zoom a lot to find what part the actual "box" is nested under.

I am quite sure #634 will be closable together with this!

**Comment by pawelszramowski** [comment]
Fixed since 1.2023.2beta2. The name is correct too since 1.2023.2beta3. Thanks!
